src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -1027,12 +1044,28 @@ private boolean addEntry(final Map<String, List<MapEntry>> entryMap, final Strin
      */
     private Map<String, Map<String, String>> loadAliases(final ResourceResolver resolver) {
         final Map<String, Map<String, String>> map = new ConcurrentHashMap<>();
-        final String queryString = "SELECT sling:alias FROM nt:base WHERE sling:alias IS NOT NULL";
-        final Iterator<Resource> i = resolver.findResources(queryString, "sql");
-        while (i.hasNext()) {
-            final Resource resource = i.next();
-            loadAlias(resource, map);
-        }
+        String queryString = "SELECT sling:alias FROM nt:base WHERE sling:alias IS NOT NULL";
+		if (this.factory.isForceNoAliasTraversal()) {
+			queryString += " option(traversal fail)";
```

Comment:
Does this introduce a dependency on a certain version of Oak?

Reply:
Yes this option was added in https://issues.apache.org/jira/browse/OAK-4888 - this is the reason why I made this configurable. Afair (to be confirmed) the query will fail with corresponding exception if the queryengine doesn't understand the option.

Reply:
Quick note as @DominikSuess pinged me offline. I wonder if it might an interesting idea to do a {{EXPLAIN SELECT sling:alias .... }} first and see that {{/\* travers}} doesn't show up in the plan.

That should work across oak versions afaict (I just tested with oak 1.2.2)

Reply:
@catholicon thx for the suggestion, it's a nice attempt but I assume it's a rather academical case and textoutput also is some kind of implementation detail that might more easily change - @rombert  I just fixed the exception handling for the non traverse option - I leave it up to you to answer the question if we should go for the more backward compatible option which requires text parsing or the option I provided which is "configurable".